agent/exec/container/adapter.go

```diff
@@ -61,15 +62,28 @@ func (c *containerAdapter) pullImage(ctx context.Context) error {
 
 	dec := json.NewDecoder(rc)
 	m := map[string]interface{}{}
+	spamLimiter := rate.NewLimiter(rate.Every(1000*time.Millisecond), 1)
+
 	for {
 		if err := dec.Decode(&m); err != nil {
 			if err == io.EOF {
 				break
 			}
 			return err
 		}
-		// TODO(stevvooe): Report this status somewhere.
-		logrus.Debugln("pull progress", m)
+		if spamLimiter.Allow() {
+			if progress, ok := m["progressDetail"].(map[string]interface{}); ok {
+				// only post pull progress to log every couple of seconds
+				log.G(ctx).WithFields(logrus.Fields{
+					"id":      m["id"],
+					"status":  m["status"],
+					"current": progress["current"],
+					"total":   progress["total"],
+				}).Debug("pull in progress")
```

Comment:
I think it would be cleaner to convert the `current` and `total` values to integers so they don't show up in scientific notation. They were integers to start out with, but when you decode a JSON number into `interface{}` it ends up as a float (because the decoder doesn't know they are meant to be integers).

Actually, a cleaner approach than converting to an integer would be to set the `UseNumber` option of the JSON decoder:

https://golang.org/pkg/encoding/json/#Decoder.UseNumber

This preserves the actual string of digits that was put into the JSON, and you can call `String` to get them (which I think logrus will do automatically when it prints the value).